### DISCUSSION TITLE
How to check/uncheck context menu items?

### DISCUSSION THREAD

**Discussion by myfix16** [HavenDV/H.NotifyIcon#62] [Q&A] [answered]
Hi,
Is there a way to check/uncheck context menu items as shown below?
<img width="155" alt="image" src="https://user-images.githubusercontent.com/61649477/211394455-f0977eb9-897c-4d17-9cab-da612a72fd48.png">

**Comment by HavenDV** [comment, marked as answer]
Do you mean console application or specific platform?

**Reply by myfix16** [reply]
In my case, Windows App SDK 1.2, but I also wonder whether it's available in all platforms.

**Reply by HavenDV** [reply]
It all depends on the specific platform and the selected ContextMenuMode for WinUI. For WPF and ContextMenuMode="SecondWindow" you can use normal xaml bindings to do this, or explicit assignment to the element via x:Name.
For the default WinUI ContextMenu (PopupMenu) there is currently no way to manage this data https://github.com/HavenDV/H.NotifyIcon/issues/8
But this can be easily added with the help of PR, this code is separate and there is not much of it:
https://github.com/HavenDV/H.NotifyIcon/tree/master/src/libs/H.NotifyIcon/PopupMenus
https://github.com/HavenDV/H.NotifyIcon/blob/aa3725bbbb808af1da1ec96f64a13a98486edcc6/src/libs/H.NotifyIcon.Shared/TaskbarIcon.ContextMenu.cs#L110-L124

**Reply by AkazaRenn** [reply]
Hi, I'm new to XAML and am currently developing a similar thing, but got into a problem, may I ask how you ended up with the implementation?

I was planning to change the icon of the items toggled between ✅ and ❎, but didn't find a way to do so. I also tried using binding but it didn't work, seems to be because my tray icon is saved in a `ResourceDictionary`, following the sample for `WinUI.Windowless`.

Do you mind sharing on how your solution looks like?

**Reply by HavenDV** [reply]
> Hi, I'm new to XAML and am currently developing a similar thing, but got into a problem, may I ask how you ended up with the implementation?
> 
> I was planning to change the icon of the items toggled between ✅ and ❎, but didn't find a way to do so. I also tried using binding but it didn't work, seems to be because my tray icon is saved in a `ResourceDictionary`, following the sample for `WinUI.Windowless`.
> 
> Do you mind sharing on how your solution looks like?

It's cool that you thought of something like this, but I'm not sure if it's feasible without changes to the library. I could help you if you could provide a PR with a test application according to your ideas, and I would help with the implementation of this at the library level, if possible.

**Reply by AkazaRenn** [reply]
@HavenDV Thank you for your reply! As suggested I've created a PR for you to check my needs here at: https://github.com/HavenDV/H.NotifyIcon/pull/67
